typescript/README.md

```diff
@@ -1 +1,178 @@
 # TypeScript Best Practices
+
+1. **Consistent Coding Style**:
+   - **Explanation**: Use a linter like ESLint to maintain a consistent coding style.
+      - @typescript-eslint/parser
+      - @typescript-eslint/eslint-plugin
+      - eslint-config-airbnb-typescript
+
+   Configure ESLint: Create or modify the `.eslintrc.json` file at the root of your project with the following configuration:
+
+   ```json
+    {
+        "extends": [
+            "airbnb-typescript"
+        ],
+        "parser": "@typescript-eslint/parser",
+        "parserOptions": {
+            "project": "./tsconfig.json"
+        },
+        "plugins": [
+            "@typescript-eslint"
+        ],
+        "rules": {
+            "semi": ["error", "always"],
+            "quotes": ["error", "single"],
+            "no-unused-vars": "error",
+            "eqeqeq": ["error", "always"],
+            "curly": ["error", "all"],
+            "no-console": "warn",
+            "consistent-return": "error",
+            "prefer-const": "error"
+        }
+    }
+    ```
+
+   This configuration allows you to benefit from Airbnb's best practices while integrating TypeScript specifics.
+
+2. **Type Annotations**:
+   - **Explanation**: Provide type annotations for function arguments and return values to ensure type safety.
+   ```typescript
+   function greet(name: string): string {
+     return `Hello, ${name}`;
+   }
+   ```
+
+3. **Interfaces**:
+   - **Explanation**: Prefer interfaces over type aliases for defining object shapes.
```

Comment:
personally I prefer using type until I need to "extends" then I use interface.
type is more intuitive to define type.
interface is more about contract that an object must adhere to or protocol.